app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -482,6 +510,16 @@ def construct_repo_from_params(repo_params) # rubocop:disable Metrics/AbcSize
       root.http_proxy_id = repo_params[:http_proxy_id] if repo_params.key?(:http_proxy_id)
       root.os_versions = repo_params.fetch(:os_versions, []) if root.yum?
 
+      if root.generic?
+        generic_remote_options = {}
+        RepositoryTypeManager.repository_types.each do |_, type|
+          type.generic_remote_options.each do |option|
```

Comment:
i wonder if we should have a RepositoryTypeManager.generic_remote_options  method, to simplify the 3 places this code appears

Reply:
I agree!

Reply:
@ianballou @jlsherrill Would this method return the equivalent of `type.generic_remote_options`?

Reply:
it would collect all the generic_remote_optiosn for all the repository_types and return them